### PR TITLE
Loki: add helper function to handle instant/range queries

### PR DESCRIPTION
**Pull request by gabor**
in the `LokiQuery` structure, before `8.4.0` we used the `.instant` and `.range` attributes to handle the two query-modes.
with `8.4.0` we are starting to migrate to the `.queryType` attribute. but, we need to be able to run queries stored in older grafana version's dashboards, so we need to be able to read both the old and new attribute(s).

this pull-request adds a helper function to take a Loki object, and normalize it.. meaning it will only contain the new `.queryType` attribute, it will not contain `.range`, it will not contain `.instant`, and it will calculate `.queryTyep` in a backward-compatible way.

i'm trying to make small pull request, so there is no code that uses this new function in this pull request. it will be used in the next pull request, that enables backend-mode.

how to test:
- if the CI tests say it is OK, it is OK 😁

### CODE REVIEW

File: public/app/plugins/datasource/loki/query_utils.test.ts [hunk 1]
```diff
@@ -61,3 +62,39 @@ describe('getHighlighterExpressionsFromQuery', () => {
...
+
+  it('handles new+old style instant', () => {
+    expectNormalized({ instant: true, range: false, queryType: LokiQueryType.Range }, LokiQueryType.Range);
+  });
+
+  it('handles new+old style range', () => {
+    expectNormalized({ instant: false, range: true, queryType: LokiQueryType.Instant }, LokiQueryType.Instant);
+  });
+
+  it('handles new<>old conflict (new wins), range', () => {
```
Comment (gabor) [inline public/app/plugins/datasource/loki/query_utils.test.ts:93]: in this approach, if new-style and old-style prop both exist, and say different things, new-style prop wins. i think this is the correct thing to do, but not 100% sure.

File: public/app/plugins/datasource/loki/query_utils.ts [hunk 2]
```diff
@@ -71,3 +72,29 @@ export function queryHasPipeParser(expr: string): boolean {
...
 }
+
+// we are migrating from `.instant` and `.range` to `.queryType`
+// the problem is, some old queries have `.instant` and `.range`,
+// and do not have `.queryType`, and we have to support those.
+// this function looks at all those properties, and returns
+// a new query object that:
+// - has `.queryType`
+// - does not have `.instant`
+// - does not have `.range`
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/query_utils.ts:83]:
```suggestion
// we are migrating from `.instant` and `.range` to `.queryType`
// this function returns a new query object that:
// - has `.queryType`
// - does not have `.instant`
// - does not have `.range`
```

**Review by ivanahuckova — APPROVED** [review]
LGTM, I have left suggestion to shorten the comment as I think that we don't need that much context in code, but leaving the decision up to you. :)

File: public/app/plugins/datasource/loki/query_utils.ts (hunk 2 above)
Comment (gabor) [reply public/app/plugins/datasource/loki/query_utils.ts]: sure!

**Pull request merged**